lua/lsp/null-ls/init.lua

```diff
@@ -0,0 +1,62 @@
+local M = {}
+
+local providers_by_ft = {}
+
+local function list_provider_names(providers, options)
+  options = options or {}
+  local names = {}
+
+  local filter = options.filter or "supported"
+  for name, _ in pairs(providers[filter]) do
+    table.insert(names, name)
+  end
+
+  return names
+end
+
+function M.list_provider_names(filetype, options)
+  local names = {}
+
+  for _, providers in pairs(providers_by_ft[filetype]) do
```

Comment:
can we check for `providers_by_ft[filetype]`not being nil 🥺

Reply:
Don't we have the guarantee that it can't be `nil` because the setup function would be called before ?